pysr/regressor_sequence.py

```diff
@@ -0,0 +1,461 @@
+            For multiple output equations, you must pass a list of indices
+            in the same order.
+        extra_predictions : int
+            If you want to predict more than one step into the future, specify
+            how many extra predictions you want. For example, if `extra_predictions=2`,
+            the model will predict the next two time points after the last time point
+            in `X`.
+
+        Returns
+        -------
+        x_predicted : ndarray of shape (n_samples, n_features)
+            Values predicted by substituting `X` into the fitted sequence symbolic
+            regression model.
+
+        Raises
+        ------
+        ValueError
+            Raises if the `best_equation` cannot be evaluated.
+        """
+        X = self._validate_data(X)
+        _check_assertions(X, recursive_history_length=self.recursive_history_length)
+        historical_X = np.lib.stride_tricks.sliding_window_view(
+            X.flatten(), self.recursive_history_length * np.prod(X.shape[1])
+        )[:: X.shape[1], :]
+        pred = self._regressor.predict(X=historical_X, index=index)
+        if extra_predictions > 0:
+            output = pred
+            previous_points = historical_X[-1]
+            # Without this, the model will re-predict the last data point
+            pred_once = self._regressor.predict(X=[previous_points], index=index)
+            previous_points = previous_points[X.shape[1] :]
+            previous_points = np.append(previous_points, pred_once)
+            previous_points = previous_points.flatten()
+            for _ in range(extra_predictions):
+                pred_once = self._regressor.predict(X=[previous_points], index=index)
+                previous_points = previous_points[X.shape[1] :]
+                previous_points = np.append(previous_points, pred_once)
+                previous_points = previous_points.flatten()
+                output = np.append(output, pred_once)
+            return output.reshape(-1, X.shape[1])
+        return pred
+
+    def from_file(
+        self,
+        cls,
+        equation_file: PathLike,
+        *pysr_args,
+        binary_operators: Optional[List[str]] = None,
+        unary_operators: Optional[List[str]] = None,
+        n_features_in: Optional[int] = None,
+        feature_names_in: Optional[ArrayLike[str]] = None,
+        selection_mask: Optional[NDArray[np.bool_]] = None,
+        nout: int = 1,
+        **pysr_kwargs
+    ):
+        return self._regressor.from_file(
+            cls,
+            equation_file,
+            *pysr_args,
+            binary_operators,
+            unary_operators,
+            n_features_in,
+            feature_names_in,
+            selection_mask,
+            nout,
+            **pysr_kwargs,
+        )
+    
+    def __repr__(self):
+        return self._regressor.__repr__().replace("PySRRegressor", "PySRSequenceRegressor")
+
+    def __getstate__(self):
```

Comment:
I don't think this function is needed? The built-in `__getstate__` should already work. The only reason PySRRegressor implements a custom one is because some of its attributes are not pickleable. But `__getstate__` works recursively so should just work fine.